src/utils/api.ts

```diff
@@ -0,0 +1,115 @@
+import config from '../config';
+import constants from '../constants';
+import axios from 'axios';
+import { IMessageTransaction, ITransactionData, IPostTransactionResponse } from '../interfaces';
+
+const path = (endpoint: string): string => config.nodes[0] + '/api' + endpoint;
+
+export const checkAccountExists = async (id: string): Promise<boolean> => {
+  const hits = await axios
+    .get(path('/delegates/' + id))
+    .then(res => Object.keys(res.data.data).length)
+    .catch(err => 0);
+
+  return hits > 0;
+};
+
+export const getTransactions = async (channel: string): Promise<IMessageTransaction[]> => {
+  const res = await axios.post(path('/transactions/search'), {
+    type: constants.messageTransaction.type,
+    typeGroup: constants.messageTransaction.typeGroup,
+    recipientId: channel
+  });
+
+  return res.data.data;
+};
+
+export const getLastMessage = async (channel: string): Promise<IMessageTransaction> => {
+  const res = await axios.post(path('/transactions/search'), {
+    type: constants.messageTransaction.type,
+    typeGroup: constants.messageTransaction.typeGroup,
+    recipientId: channel
+  });
+
+  return res.data.data[0];
+};
+
+export const fetchUsername = async (address: string): Promise<string | null> => {
+  const username = await axios
+    .get(path('/delegates/' + address))
+    .then(res => {
```

Comment:
This is inconsistent. You are using `await` but then also `then` and `catch` instead of simply using `await` and `try/catch`.

Reply:
Resolved in https://github.com/ArkEcosystem/poc-ark-messenger/pull/5